### PR TITLE
Removes legacy react contexttypes and uses formatmessage from react-intl.

### PR DESCRIPTION
**Pull request by lokesh-0813**
### Summary
Few packages which were interfacing with legacy react `contexttypes` are removed. Now the packages use an abstraction like `formatMessage` to interface with `injectIntl`.

Solves : #523

### CODE REVIEW

File: packages/terra-clinical-onset-picker/CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,10 @@ ChangeLog
 
 Unreleased
 ----------
+### Changed
+* Removed react legacy `contexttypes`.
+* Updated the component with `formatMessage` to interface with `react-intl`.
```
Comment (nramamurth) [inline packages/terra-clinical-onset-picker/CHANGELOG.md]: I think you meant to say `injectIntl` here.

Comment (lokesh-0813) [reply packages/terra-clinical-onset-picker/CHANGELOG.md]: I will have it updated. Thank you.

File: packages/terra-clinical-onset-picker/src/OnsetPicker.jsx [hunk 2]
```diff
@@ -85,6 +86,11 @@ const propTypes = {
    * Whether or not the legend is visible. Use this props to hide a legend while still creating it on the DOM for accessibility.
    */
   isLegendHidden: PropTypes.bool,
+
+  /**
+   * The intl object containing translations. This is retrieved from the context automatically by injectIntl.
```
Comment (bjankord) [inline packages/terra-clinical-onset-picker/src/OnsetPicker.jsx]: We should add the @private syntax so this doesn't show up in the generated prop-types documentation.
```suggestion
   * @private
   * The intl object containing translations. This is retrieved from the context automatically by injectIntl.
```

**Comment by bjankord** [comment]
I'm talking with the team about this update and will post back once I have some consensus on the direction we want to go with this.

**Comment by bjankord** [comment]
@lokesh-0813 Talked about this with @ryanthemanuel. The plan is to move forward with `injectIntl` in the places that you've identified where it doesn't make sense to use `<FormattedMessage />`

**Comment by lokesh-0813** [comment]
Sticking to `injectIntl` since `formattedMessage` does not make sense.

**Pull request merged**